CAIPs/caip-282.md

```diff
@@ -0,0 +1,405 @@
+// Blockchain Library will send a CAIP-25 request to establish a wallet connection
+let session = {};
+window.addEventListener("caip282:respond:350670db-19fa-4704-a166-e52e178b59d2", (event) => {
+  if (event.data.error) throw new Error(event.data.error.message);
+  session = event.data.result;
+});
+window.postMessage("message", {
+  event: "caip282:request:350670db-19fa-4704-a166-e52e178b59d2",
+  data: {
+    id: 1,
+    jsonrpc: "2.0",
+    method: "provider_authorize",
+    params: {
+      optionalScopes: {
+        eip155: {
+          scopes: ["eip155:1", "eip155:10"],
+          methods: ["eth_sendTransaction", "personal_sign"],
+          notifications: ["accountsChanged", "chainChanged"],
+        },
+      },
+      sessionProperties: {
+        expiry: "2024-06-06T13:10:48.155Z",
+      },
+    },
+  },
+});
+
+// 4. After the response was received by the Blockchain Library from the wallet
+// provider then the session is established with a sessionId matchin the UUID
+// thus signing requests can be using a CAIP-27 request to the wallet user
+let result = {};
+window.addEventListener("caip282:respond:350670db-19fa-4704-a166-e52e178b59d2", (event) => {
+  if (event.data.error) throw new Error(event.data.error.message);
+  result = event.data.result;
+});
+window.postMessage("message", {
+  event: "caip282:request:350670db-19fa-4704-a166-e52e178b59d2",
+  data: {
+    {
+      id: 2,
+      jsonrpc: "2.0",
+      method: "provider_request",
+      params: {
+        sessionId: "350670db-19fa-4704-a166-e52e178b59d2",
+        scope: "eip155:10",
+        request: {
+          method: "eth_sendTransaction",
+          params: [
+            {
+              type: "0x2",
+              nonce: "0x01",
+              value: "0x00",
+              maxFeePerGas: "0x9143798a4",
+              maxPriorityFeePerGas: "0x59682f00",
+              from: "0x43e3ca49c7be4f429abce408da6b738f879d02a0",
+              to: "0x0b2C639c533813f4Aa9D7837CAf62653d097Ff85",
+              data: "0xa9059cbb000000000000000000000000677d6d2747955ecf1e9fad3521d29512fb599e7b0000000000000000000000000000000000000000000000000de0b6b3a7640000",
+            }
+          ]
+        }
+      }
+    }
+  },
+});
+
+```
+
+Logic from the Wallet Provider:
+
+```js
+// 1. Wallet Provider sets their WalletData and then listens to promptWallet message
+// and also immediatelly posts a message with the WalletData as announceWallet type
+const data = {
+  uuid: generateUUID(); // eg. "350670db-19fa-4704-a166-e52e178b59d2"
+  name: "Example Wallet",
+  icon: "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg==";
+  rdns: "com.example.wallet";
+}
+window.addEventListener("caip282:promptWallet", (event) => {
+  // when a prompt message was received then the wallet will announces again
+  window.postMessage("message", {
+    event: "caip282:announceWallet",
+    data,
```

Comment:
why not just have `method, params` a la JSON-RPC. caip-27 nests JSON-RPC so nesting here doesn't sound too bad I guess.

Reply:
Just changed it 👍